lib/trino-plugin-toolkit/src/main/java/io/trino/plugin/base/statistics/Statistics.java

```diff
@@ -0,0 +1,239 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.plugin.base.statistics;
+
+import com.google.common.collect.ImmutableMap;
+import io.trino.spi.block.Block;
+import io.trino.spi.statistics.ColumnStatisticMetadata;
+import io.trino.spi.statistics.ColumnStatisticType;
+import io.trino.spi.type.DecimalType;
+import io.trino.spi.type.Decimals;
+import io.trino.spi.type.Type;
+
+import java.math.BigDecimal;
+import java.time.LocalDate;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Optional;
+import java.util.OptionalDouble;
+import java.util.OptionalLong;
+
+import static com.google.common.base.Verify.verify;
+import static com.google.common.collect.ImmutableMap.toImmutableMap;
+import static io.trino.plugin.base.statistics.ReduceOperator.ADD;
+import static io.trino.plugin.base.statistics.ReduceOperator.MAX;
+import static io.trino.plugin.base.statistics.ReduceOperator.MIN;
+import static io.trino.spi.type.BigintType.BIGINT;
+import static io.trino.spi.type.DateType.DATE;
+import static io.trino.spi.type.DoubleType.DOUBLE;
+import static io.trino.spi.type.IntegerType.INTEGER;
+import static io.trino.spi.type.RealType.REAL;
+import static io.trino.spi.type.SmallintType.SMALLINT;
+import static io.trino.spi.type.TinyintType.TINYINT;
+import static java.lang.Float.intBitsToFloat;
+import static java.lang.Math.toIntExact;
+
+public final class Statistics
+{
+    private Statistics() {}
+
+    public static Map<String, Map<ColumnStatisticType, Block>> createColumnToComputedStatisticsMap(Map<ColumnStatisticMetadata, Block> computedStatistics)
+    {
+        Map<String, Map<ColumnStatisticType, Block>> result = new HashMap<>();
```

Comment:
possible java streams related improvement

```java
        return computedStatistics.entrySet()
                .stream()
                .collect(collectingAndThen(
                        groupingBy(entry -> entry.getKey().getColumnName(),
                                mapping(Functions.identity(),
                                        Collectors.toUnmodifiableMap(entry -> entry.getKey().getStatisticType(), Map.Entry::getValue))),
                        ImmutableMap::copyOf));
```

I can create a PR for your branch if you want with a few functionality tests for  the method `io.trino.plugin.base.statistics.Statistics#createColumnToComputedStatisticsMap`